File: C/fizzbuzz3.c
#include <stdio.h>

#define LIMIT 100

int main()
{
	int i;

	for (i = 1; i <= LIMIT; i++) {
		switch (i % 15) {
			case 0:
				printf("FizzBuzz");
				break;
			case 3:
			case 6:
			case 9:
			case 12:
				printf("Fizz");
				break;
			case 5:
			case 10:
				printf("Buzz");
				break;
			default:
				printf("%d", i);
				break;
		}
		if (i < LIMIT) {
			printf(", ");
		}
	}
	printf("\n");
}

File: Erlang/fizzbuzz3.erl
-module(fizzbuzz3).
-export([main/0, main/1]).

-define(LIMIT, 100).

main() ->
    io:fwrite("~s~n", [lists:flatten(lists:join(", ", lists:map(
        fun(X) ->
            if X rem 15 == 0 -> "FizzBuzz";
               X rem 3 == 0 -> "Fizz";
               X rem 5 == 0 -> "Buzz";
               true -> integer_to_list(X)
            end
        end
        , lists:seq(1, ?LIMIT))))]).

main(_) -> main().
